client/data/emails/use-remove-email-forward-mutation.ts

```diff
@@ -0,0 +1,50 @@
+import { useMutation, useQueryClient } from 'react-query';
+import { useSelector } from 'react-redux';
+import wp from 'calypso/lib/wp';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { getCacheKey as getEmailAccountsQueryKey } from './use-get-email-accounts-query';
+import { getCacheKey as getEmailDomainsQueryKey } from './use-get-email-domains-query';
+import type { UseMutationOptions, UseMutationResult } from 'react-query';
+
+/**
+ * Deletes a mailbox from a Professional Email (Titan) account
+ *
+ * @param domainName The domain name of the mailbox
+ * @param mailboxName The mailbox name
+ * @param mutationOptions Mutation options passed on to `useMutation`
+ * @returns {{ data, error, isLoading: boolean, removeTitanMailbox: Function, ...}} Returns various parameters piped from `useMutation`
+ */
+export function useRemoveEmailForwardMutation(
+	domainName: string,
+	mailboxName: string,
+	mutationOptions: UseMutationOptions = {}
+): UseMutationResult< unknown, unknown, void, unknown > {
+	const queryClient = useQueryClient();
+
+	const selectedSiteId = useSelector( getSelectedSiteId );
+
+	const useEmailsQueryKey = getEmailAccountsQueryKey( selectedSiteId, domainName );
+	const useGetEmailDomainsQueryKey = getEmailDomainsQueryKey( selectedSiteId );
+
+	const suppliedOnSettled = mutationOptions.onSettled;
+
+	mutationOptions.onSettled = async ( data, error, variables, context ) => {
+		suppliedOnSettled?.( data, error, variables, context );
```

Comment:
My thinking was that we should be able to mitigate that since this function is now defined in Typescript. I realize that we don't get type checking if `useRemoveEmailForwardMutation` is called from a plain ol' JS file though.

Thinking about it now however, I kind of feel that it's a fair assumption that if `onSettled` is defined, it is a function. Seems a little bit heavy to do a runtime type check to prevent such a "silly" mistake.

We don't have a runtime type check like that in `client/data/emails/use-remove-titan-mailbox-mutation.ts` (and we didn't have it before this PR either).

Reply:
That's indeed a fair assumption!